core/src/main/java/hudson/FilePath.java

```diff
@@ -1400,17 +1402,15 @@ public String invoke(File dir, VirtualChannel channel) throws IOException {
      * @return
      *      The new FilePath pointing to the temporary directory
      * @since 1.311
-     * @see File#createTempFile(String, String)
+     * @see Files#createTempDirectory(Path, String, FileAttribute[])
      */
     public FilePath createTempDir(final String prefix, final String suffix) throws IOException, InterruptedException {
         try {
             return new FilePath(this,act(new SecureFileCallable<String>() {
                 private static final long serialVersionUID = 1L;
                 public String invoke(File dir, VirtualChannel channel) throws IOException {
-                    File f = File.createTempFile(prefix, suffix, dir);
-                    f.delete();
-                    f.mkdir();
-                    return f.getName();
+                    Path tempPath = Files.createTempDirectory(dir.toPath(), prefix, null);
```

Comment:
`dir.toPath()` may throw `InvalidPathException`. Since it is an API method, it has to be handled and wrapped to `IOException`

Reply:
Agreed - change added